roles/gateway_import_database/tasks/upgrade_gateway_database_schema.yml

```diff
@@ -1,8 +1,17 @@
 ---
 # This task upgrade the gateway database version.
 
+- name: Check if Gateway Database Schema needs update
+  no_log: true # don't log decrypted passwords in responses
+  shell:
+    cmd: sshpass -p '{{ ansible_password }}' ssh ssgconfig@{{ inventory_hostname }} 'sudo -u layer7 /opt/SecureSpan/Appliance/libexec/ssgconfig_launch -databaseUgrade' 
+  register: update
+  changed_when: false
+  delegate_to: localhost
+
 - name: upgrading Gateway Database Schema after import database
   no_log: true # don't log decrypted passwords in responses
+  when:  '"Database upgrade not required" not in update.stdout'
```

Comment:
we should find the exact message when upgrade is needed, any other condition is skipped.

Reply:
Here is what it says when upgraded is needed:
[root@jason-ssg94d tmp]# /opt/SecureSpan/Appliance/libexec/ssgconfig_launch -databaseUgrade
Gateway Database Upgrader.
Enter 'quit' to exit at any time.

Database upgrade is required.

 Software version : 10.0.00
 Database version : 9.4.00

Perform upgrade? [No]:

Reply:
I guess we are going with the safer option.  Once the database schema has been upgraded, it cannot be upgraded again.  Please also mention this in the readme for this role if it is not already there.